src/library-authoring/create-container/CreateContainerModal.test.tsx

```diff
@@ -0,0 +1,241 @@
+    axiosMock.onPatch(collectionItemsUrl).reply(200);
+
+    renderComponent();
+
+    // Fill in the form
+    const nameInput = screen.getByLabelText(/name/i);
+    await userEvent.type(nameInput, 'Test Container');
+
+    // Submit the form
+    const createButton = screen.getByRole('button', { name: /create/i });
+    await userEvent.click(createButton);
+
+    // Verify API calls
+    await waitFor(() => {
+      // Verify container creation call
+      expect(axiosMock.history.post).toHaveLength(1);
+      expect(axiosMock.history.post[0].url).toMatch(/\/api\/libraries\/.*\/containers/);
+      expect(JSON.parse(axiosMock.history.post[0].data)).toEqual({
+        can_stand_alone: false,
+        container_type: 'section',
+        display_name: 'Test Container',
+      });
+
+      // Verify collection linking call
+      expect(axiosMock.history.patch).toHaveLength(1);
+      expect(axiosMock.history.patch[0].url).toBe(collectionItemsUrl);
+      expect(JSON.parse(axiosMock.history.patch[0].data)).toEqual({
+        usage_keys: ['lct:org:libId:collection:1'],
+      });
+
+      expect(mockNavigateTo).toHaveBeenCalledWith({ containerId: 'lct:org:libId:collection:1' });
+      expect(mockShowToast).toHaveBeenCalled();
+    });
+  });
+
+  it('links container to section when inside a section', async () => {
+    // Mock the container creation API
+    axiosMock.onPost(/\/api\/libraries\/.*\/containers/).reply(200, {
+      id: 'lct:org:libId:section:1',
+      displayName: 'Test Container',
+    });
+
+    // Mock the container linking API
+    const containerChildrenUrl = getLibraryContainerChildrenApiUrl('lct:org:lib:unit:test-container', false);
+    axiosMock.onPost(containerChildrenUrl).reply(200);
+
+    // Override the default context for section test
+    jest.mocked(useLibraryContext).mockReturnValue({
+      ...defaultLibraryContext,
+      collectionId: undefined,
+      containerId: 'lct:org:lib:unit:test-container',
+    });
+
+    // Update the useLibraryRoutes mock for section test
+    mockUseLibraryRoutes.mockReturnValue({
+      navigateTo: mockNavigateTo,
+      insideCollection: false,
+      insideSection: true,
+      insideSubsection: false,
+    });
+
+    renderComponent();
+
+    // Fill in the form
+    const nameInput = screen.getByLabelText('Name your section');
+    await userEvent.type(nameInput, 'Test Container');
+
+    // Submit the form
+    const createButton = screen.getByRole('button', { name: /create/i });
+    await userEvent.click(createButton);
+
+    // Verify API calls
+    await waitFor(() => {
+      // Verify container creation call
+      expect(axiosMock.history.post[0].url).toMatch(/\/api\/libraries\/.*\/containers/);
+      expect(JSON.parse(axiosMock.history.post[0].data)).toEqual({
+        can_stand_alone: false,
+        container_type: 'section',
+        display_name: 'Test Container',
+      });
```

Comment:
@DanielVZ96 It's recommended to [avoid multiple assertions](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback) in a single `waitfor`